### PR TITLE
feat: add support for mounting secrets

### LINKED ISSUE #187: Support using secrets
### TL;DR

It would be great if this action supports passing the secret as an environment variable or mounting the secret as a volume.

### Design

**Action YAML**

```yaml
steps:
- uses: actions/checkout@v2
- id: deploy
  uses: google-github-actions/deploy-cloud-functions@main
  with:
    name: my-function
    runtime: nodejs10
    credentials: ${{ secrets.gcp_credentials }}
    secrets: |
      ENV_VAR=SECRET:VERSION
      ENV_VAR=projects/PROJECT_ID/secrets/SECRET:VERSION
      ENV_VAR=projects/PROJECT_ID/secrets/SECRET/versions/VERSION
      /SECRET_PATH=SECRET:VERSION
      /MOUNT_PATH:/SECRET_PATH=SECRET:VERSION

```

**Resources**

- [Using secrets &nbsp;|&nbsp; Cloud Functions Documentation &nbsp;|&nbsp; Google Cloud](https://cloud.google.com/functions/docs/configuring/secrets)
- [gcloud beta functions deploy &nbsp;|&nbsp; Cloud SDK Documentation &nbsp;|&nbsp; Google Cloud](https://cloud.google.com/sdk/gcloud/reference/beta/functions/deploy#--set-secrets)

### PR DESCRIPTION
**Pull request by sethvargo**
Adds support for mounting secrets via environment variables or via the filesystem.

- Fixes #187
- Closes #188

### CODE REVIEW

File: src/secret.ts [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+      // <p>/<s>
+      case 2: {
+        this.project = refParts[0];
+        this.name = refParts[1];
+        this.version = 'latest';
+        break;
+      }
+      default: {
+        throw new TypeError(
+          `Failed to parse secret reference "${s}": unknown format`,
```
Comment (bharathkkb) [inline src/secret.ts:67]: Can we suggest the `projects/p/secrets/s/versions/v` format

File: src/main.ts [hunk 2]
```diff
@@ -127,19 +140,56 @@ async function run(): Promise<void> {
...
+      for (const [key, value] of Object.entries(secretVols)) {
+        const mountPath = posix.dirname(key);
+        const pth = posix.basename(key);
+
+        const secretRef = new SecretName(value);
+        secretVolumes.push({
+          mountPath: mountPath,
+          projectId: secretRef.project,
+          secret: secretRef.name,
+          versions: [
```
Comment (bharathkkb) [inline src/main.ts:176]: Does API support mounting multiple versions?

**Review by bharathkkb — COMMENTED** [review]
LGTM

File: src/main.ts (hunk 2 above)
Comment (sethvargo) [reply src/main.ts:176]: No, which is why it's weird that it's an array. The API is... strange.

File: src/secret.ts (hunk 1 above)
Comment (bharathkkb) [reply src/secret.ts:67]: https://cloudfunctions.googleapis.com/$discovery/rest?version=v1 for more details

File: src/main.ts (hunk 2 above)
Comment (bharathkkb) [reply src/main.ts:176]: https://cloudfunctions.googleapis.com/$discovery/rest?version=v1 for more details

File: src/secret.ts (hunk 1 above)
Comment (sethvargo) [reply src/secret.ts]: Done

**Review by bharathkkb — APPROVED** [review]

**Pull request merged**